src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -467,7 +467,9 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
       val parameter = primaryConstructor?.parameter(constructorParameter)
       requireNotNull(parameter) {
         "no such constructor parameter '$constructorParameter' to delegate to for type '$name'" }
-      addSuperinterface(parameter!!.type, CodeBlock.of(parameter.name))
+      require(parameter!!.type == superinterface) {
```

Comment:
This check will break things like:

```kotlin
class A : Runnable
class B(a: A) : Runnable by a
```

Reply:
I think about these things after I push, my bad for making this painful o_0

Reply:
No worries! By the way, I think it would be great to have a unit test for this specific case

Reply:
will do!